src/ImmersedBoundaries/grid_fitted_immersed_boundaries.jl

```diff
@@ -76,8 +76,9 @@ function ImmersedBoundaryGrid(grid, ib::AbstractGridFittedBottom{<:OffsetArray})
 end
 
 @inline function immersed_cell(i, j, k, underlying_grid, ib::GridFittedBottom)
-    z = znode(c, c, c, i, j, k, underlying_grid)
-    return @inbounds z < ib.bottom_height[i, j]
+    z = znode(c, c, f, i, j, k+1, underlying_grid)
+    h = @inbounds ib.bottom_height[i, j]
+    return z <= h
```

Comment:
is it `z <= h` or `z < h`? 

Might want to change

https://github.com/CliMA/Oceananigans.jl/blob/1b4ac9b4467eced63ed47b49e9fe2cc12a5658c2/src/ImmersedBoundaries/partial_cell_immersed_boundaries.jl#L52

to be the _exactly_ the same too.